Lib9c/Action/TransferAssets.cs

```diff
@@ -0,0 +1,170 @@
+using Bencodex;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Nekoyume.Model.State;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using Nekoyume.Model;
+using Serilog;
+
+namespace Nekoyume.Action
+{
+    /// <summary>
+    /// Hard forked at https://github.com/planetarium/lib9c/pull/636
+    /// Updated at https://github.com/planetarium/lib9c/pull/957
+    /// </summary>
+    [Serializable]
+    [ActionType("transfer_assets")]
+    public class TransferAssets : ActionBase, ISerializable, ITransferAssets
+    {
+        private const int MemoMaxLength = 80;
+
+        public TransferAssets()
+        {
+        }
+
+        public TransferAssets(Address sender, Dictionary<Address, FungibleAssetValue> map, string memo = null)
+        {
+            Sender = sender;
+            Map = map;
+
+            CheckMemoLength(memo);
+            Memo = memo;
+        }
+
+        protected TransferAssets(SerializationInfo info, StreamingContext context)
+        {
+            var rawBytes = (byte[])info.GetValue("serialized", typeof(byte[]));
+            Dictionary pv = (Dictionary) new Codec().Decode(rawBytes);
+
+            LoadPlainValue(pv);
+        }
+
+        public Address Sender { get; private set; }
+        public Dictionary<Address, FungibleAssetValue> Map { get; private set; }
```

Comment:
If we want to keep this `Dicionary<>`'s order, how about using `List<Tuple<Address, FungibleAssetValue>>` instead?

Reply:
or, preserving the ordered internal field seems helpful to avoid re-ordering at least.